src/Drivers/nlpSparse_ex9_driver.cpp

```diff
@@ -0,0 +1,97 @@
+#include "nlpSparse_ex9.hpp"
+#include "hiopNlpFormulation.hpp"
+#include "hiopAlgFilterIPM.hpp"
+
+
+using namespace hiop;
+
+#ifdef HIOP_USE_MAGMA
+#include "magma_v2.h"
+#endif
+
+#include <cstdlib>
+#include <string>
+
+/**
+ * Driver for example 8 that illustrates the use of hiop::hiopAlgPrimalDecomposition 
+ * 
+ * @note This example is built only when HIOP_USE_MPI is enabled during cmake build
+ * and require at least two MPI ranks in MPI_COMM_WORLD.
+ *
+ */
+
+//
+//todo: add -selfcheck option (see other drivers) and add the driver to cmake tests
+//
+
+using namespace hiop;
+
+
+int main(int argc, char **argv)
+{
+  int rank=0;
+#ifdef HIOP_USE_MPI
+  MPI_Init(&argc, &argv);
+  int comm_size;
+  int ierr = MPI_Comm_size(MPI_COMM_WORLD, &comm_size); assert(MPI_SUCCESS==ierr);
+  ierr = MPI_Comm_rank(MPI_COMM_WORLD, &rank); assert(MPI_SUCCESS==ierr);
+  double t3 =  MPI_Wtime(); 
+  double t4 = 0.; 
+#endif
+
+#ifdef HIOP_USE_MAGMA
+  magma_init();
+#endif
+  int nx = 1000;
+  int S = 1920;
+  int nS = 5; 
+  double x[nx+S*nx];
+
+
+  Ex9 nlp_interface(nx,S,nS);
+  hiopNlpSparse nlp(nlp_interface);
+  //nlp.options->SetStringValue("duals_update_type", "linear"); // "lsq" or "linear" --> lsq hasn't been implemented yet.
+                                                            // it will be forced to use "linear" internally.
+  //nlp.options->SetStringValue("duals_init", "lsq"); // "lsq" or "zero"
+  nlp.options->SetStringValue("compute_mode", "cpu");
+  //nlp.options->SetStringValue("compute_mode", "hybrid");
+  nlp.options->SetStringValue("KKTLinsys", "xdycyd");
+  //nlp.options->SetStringValue("KKTLinsys", "full");
+  //nlp.options->SetStringValue("write_kkt", "yes");
```

Comment:
Remove commented out options and document options for user to play with (if any).

Reply:
For the sake of the problem, no options are currently needed. Some of them were used during debugging. I'm not sure how many we want to keep.

Reply:
@jwang125, it is your call how many options to keep. The `compute_mode` one is I would suggest to keep in all examples as we are working with more and more GPU code.